ai_services/anomaly_detection/data_preprocessing_examples/oci_data_flow_based_examples/example_code/ad_utils.py

```diff
@@ -63,28 +73,102 @@ def _create_model_(self, project_id, compartment_id, data_asset_id,
             f"Error creating model: {create_model_response.text}"
         return create_model_response.data.id
 
+    def create_detect_anomalies_job(self, compartment_id, model_id,
+                                    data_asset_detail,
+                                    output_path) -> DetectAnomalyJob:
+
+        input_details = ObjectListInputDetails(
+            object_locations=[ObjectLocation(
+                namespace_name=data_asset_detail['namespace'],
+                bucket_name=data_asset_detail['bucket'],
+                object_name=data_asset_detail['object'])])
+        output_details = ObjectStoreOutputDetails(
+            namespace_name=data_asset_detail['namespace'],
+            bucket_name=output_path,
+            prefix="model1")
+        create_detect_anomaly_job_details = CreateDetectAnomalyJobDetails(
+            compartment_id=compartment_id,
+            display_name="e2e_template_test_job",
+            model_id=model_id,
+            input_details=input_details,
+            output_details=output_details)
+        print(create_detect_anomaly_job_details)
+        response = self.ad_client.create_detect_anomaly_job(
+            create_detect_anomaly_job_details)
+        print(response)
+        assert response.status == 200, \
+            f"Error detecting anomalies: {response.status}"
+        return response.data
+
+    def infer(self, compartment_id:str, model_id:str, staging_details, output_path)\
+            -> int:
+        jobs = []
+        for data_asset_detail in staging_details:
+            jobs.append(self.create_detect_anomalies_job(compartment_id,
+                                                         model_id,
+                                                         data_asset_detail,
+                                                         output_path))
+        retries = 0
+        while jobs and retries < 10:
+            retries = retries + 1
+            for job in list(jobs):
```

Comment:
I believe it will be problematic when you iterate a list while removing element from it
```
>>> x = [1,2,3,4,5]
>>> for y in x:
...     print(y)
...     x.remove(y)
...
1
3
5
```

Reply:
here I m making a copy of the list using the list() call, so there will be no issues. Pls see https://stackoverflow.com/questions/6022764/python-removing-list-element-while-iterating-over-list